src/index.ts

```diff
@@ -154,59 +141,60 @@ client.on('messageCreate', async (message) => {
-        (oldState.serverMute && !newState.serverMute) ||
-          (!oldState.serverMute && newState.serverMute)
-      ];
-      const isSwitchingChannel = Boolean(oldState.channel && newState.channel);
-      const isJoiningChannel = oldState.channel === null && newState.channel !== null;
-      if (
-        DONT_INTRO.every((condition) => condition === false) &&
-        (isJoiningChannel || isSwitchingChannel)
-      ) {
-        // Play a clip based on the username
-        if (discordUserAnnouncementDictionary[usernameNoHash]) {
-          playClip(
-            `${PATH_TO_CLIPS}${discordUserAnnouncementDictionary[usernameNoHash]}`,
-            channel,
-            audioPlayer
-          );
-        } else {
-          if (!botUsernames.includes(username)) {
-            console.log('Unhandled user joined a voice channel. Announcing...');
-            annouceUnhandledUser(channel, audioPlayer, usernameNoHash);
-          }
+  // Set the channel for the bot to join
+  channel = newState.channel;
+  if (channel) {
+    // Grab the username of the user who joined
+    const username = newState?.member?.user.tag as string;
+    const usernameNoHash = username?.split('#')[0] ?? '';
+    if (!oldState.streaming && newState.streaming) {
+      annouceUserIsStreaming(channel, audioPlayer, usernameNoHash);
+    }
+    // The voiceStateUpdate callback is triggered for a variety of reasons, but we only care about some of them for intros.
+    const DONT_INTRO = [
+      (oldState.streaming && !newState.streaming) || (!oldState.streaming && newState.streaming),
+      (oldState.selfDeaf && !newState.selfDeaf) || (!oldState.selfDeaf && newState.selfDeaf),
+      (oldState.selfMute && !newState.selfMute) || (!oldState.selfMute && newState.selfMute),
+      (oldState.serverDeaf && !newState.serverDeaf) ||
+        (!oldState.serverDeaf && newState.serverDeaf),
+      (oldState.serverMute && !newState.serverMute) || (!oldState.serverMute && newState.serverMute)
+    ];
+    const isSwitchingChannel = Boolean(oldState.channel && newState.channel);
+    const isJoiningChannel = oldState.channel === null && newState.channel !== null;
+    if (
+      DONT_INTRO.every((condition) => condition === false) &&
+      (isJoiningChannel || isSwitchingChannel)
+    ) {
+      // Look for a folder for the user. If there is one, play a random clip from it.
+      if (!botUsernames.includes(username)) {
+        const success = await playRandomClipFromFolder(
+          `${EventFiles.DIS_USER_ENTER}${usernameNoHash}`,
+          channel,
+          audioPlayer
+        );
+
+        // If no clip was found, announce the user using TTS.
+        if (!success) {
+          console.log('Unhandled user joined a voice channel. Announcing...');
+          // TODO: this broke, think something got screwed up here with dependency updates looking at error.
+          // await announceUnhandledUser(channel, audioPlayer, usernameNoHash);
```

Comment:
This spits out a huge error now and crashes the bot, didn't touch anything in the function other than correcting the function name